module/zfs/abd.c

```diff
@@ -247,9 +346,11 @@ static inline abd_t *
 abd_alloc_struct(size_t chunkcnt)
 {
 	size_t size = offsetof(abd_t, abd_u.abd_scatter.abd_chunks[chunkcnt]);
-	abd_t *abd = kmem_alloc(size, KM_PUSHPAGE);
+	abd_t *abd = kmem_zalloc(size, KM_PUSHPAGE);
 	ASSERT3P(abd, !=, NULL);
 	ABDSTAT_INCR(abdstat_struct_size, size);
+	abd->abd_create_time = gethrtime();
+	mutex_init(&abd->abd_mutex, NULL, MUTEX_DEFAULT, NULL);
```

Comment:
It is more that when you use `kmem_cache()` it calls the `init` on a new allocation only, but between use, and re-use, the mutex is not freed and re-initialised. The same mutex is used with many `znodes` for example, like a floozy.